code/modules/mob/mob_helpers.dm

```diff
@@ -584,11 +584,19 @@ var/list/intents = list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM)
 					break
 			if(newname)
 				break	//That's a suitable name!
-			to_chat(src, "Sorry, that [role]-name wasn't appropriate, please try another. It's possibly too long/short, has bad characters or is already taken.")
+			to_chat(src, "Sorry, that [role] name wasn't appropriate, please try another. It's possibly too long/short, has bad characters or is already taken.")
 
 		if(!newname)	//we'll stick with the oldname then
 			return
 
+		if(ishuman(src))
+			var/mob/living/carbon/human/H = src

```

Comment:
This is an egregious violation of OOP. Don't do this.

Reply:
had no clue what else to do to cast it.

Reply:
It needs to be more OOP; this should be under the human level proc, not at the mob level.

Reply:
is there even a human level proc at the moment?